bridge/setu/broadcaster/broadcaster.go

```diff
@@ -0,0 +1,133 @@
+package broadcaster
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	cliContext "github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bor "github.com/maticnetwork/bor"
+	"github.com/maticnetwork/bor/core/types"
+	authTypes "github.com/maticnetwork/heimdall/auth/types"
+	"github.com/maticnetwork/heimdall/bridge/setu/util"
+	"github.com/maticnetwork/heimdall/helper"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// TxBroadcaster uses to broadcast transaction to each chain
+type TxBroadcaster struct {
+	logger log.Logger
+
+	cliCtx cliContext.CLIContext
+
+	heimdallMutex sync.Mutex
+	maticMutex    sync.Mutex
+}
+
+// NewTxBroadcaster creates new broadcaster
+func NewTxBroadcaster(cdc *codec.Codec) *TxBroadcaster {
+	cliCtx := cliContext.NewCLIContext().WithCodec(cdc)
+	cliCtx.BroadcastMode = client.BroadcastSync
+	cliCtx.TrustNode = true
+
+	txBroadcaster := TxBroadcaster{
+		logger: util.Logger().With("module", "txBroadcaster"),
+		cliCtx: cliCtx,
+	}
+
+	return &txBroadcaster
+}
+
+// BroadcastToHeimdall broadcast to heimdall
+func (tb *TxBroadcaster) BroadcastToHeimdall(msg sdk.Msg) error {
```

Comment:
Can we test two tasks calling `BroadcastToHeimdall` without any error parallelly?

Reply:
done. added a test case for the same.